client-course-schedulizer/src/utilities/helpers/caseFunctions.ts

```diff
@@ -27,12 +27,40 @@ const thursReg = RegExp("[Tt][Hh]|[Rr]");
 const friReg = RegExp("[Ff]");
 const satReg = RegExp("[Ss](?![Uu])");
 
-export const startTimeCallback = (value: string, { firstMeeting }: CaseCallbackParams) => {
-  firstMeeting.startTime = startTimeCase(value);
+export const startTimeCallback = (value: string, { meetings }: CaseCallbackParams) => {
+  const startTimes = value.split("\n");
+  let i = 0;
+  startTimes.forEach((startTime) => {
+    if (meetings.length <= i) {
+      // If there aren't enough meetings, create a new one
+      meetings.push({
+        days: [],
+        duration: 0,
+        location: { building: "", roomCapacity: 0, roomNumber: "" },
+        startTime: "",
+      });
+    }
+    meetings[i].startTime = startTimeCase(startTime);
+    i += 1;
+  });
 };
 
-export const locationCallback = (value: string, { firstMeeting }: CaseCallbackParams) => {
-  [firstMeeting.location.building, firstMeeting.location.roomNumber] = locationCase(value);
+export const locationCallback = (value: string, { meetings }: CaseCallbackParams) => {
+  const locations = value.split("\n");
+  let i = 0;
+  locations.forEach((location) => {
+    if (meetings.length <= i) {
+      // If there aren't enough meetings, create a new one
+      meetings.push({
+        days: [],
+        duration: 0,
+        location: { building: "", roomCapacity: 0, roomNumber: "" },
+        startTime: "",
+      });
+    }
```

Comment:
This is repeated code, so it's probably best to make a function for it

Reply:
Also, replace the repeated code in the other callbacks with a call to the new function.

Reply:
You should put the `split()` line in the function too.

Reply:
I tried this in cac67a9, but I think in certain ways, it makes things more confusing, so let me know if you want me to revert the change or have suggestions on how it could be done better.

Reply:
2453811 makes this less of an issue